test/FnameResolver/FnameResolver.t.sol

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "forge-std/Test.sol";
+
+import "../TestConstants.sol";
+import {FnameResolverTestSuite} from "./FnameResolverTestSuite.sol";
+import {FnameResolver} from "../../src/FnameResolver.sol";
+
+/* solhint-disable state-visibility */
+
+contract FnameResolverTest is FnameResolverTestSuite {
```

Comment:
We should also add a js test to make sure the full resolution flow works via existing ENS libraries (ethers/viem)